### DISCUSSION TITLE
How to turn Flowise into an API service?

### DISCUSSION THREAD

**Discussion by ops-004** [FlowiseAI/Flowise#1212] [Q&A]
Hello, I would like to use Flowise as an API service rather than just a prototyping tool. As a remote API service, I need to address two core issues:

1. Single point of failure
2. Concurrent execution of multiple tasks

My current idea is to deploy multiple Flowise nodes on separate physical machines and use nginx for load balancing. This should help solve the issues of single point of failure and concurrent task execution.

However, there is still uncertainty with this approach. For example, if Flowise-1 node creates a flow or makes other configuration changes, will Flowise-2 be able to share those updates?

I have reviewed some information but still can't resolve my doubts. So, I would like to confirm with you whether Flowise currently supports deploying on multiple machines with multiple nodes to build a highly available distributed API application. This is extremely important for me.

I also have some other ideas and questions:

1. Does the Flowise API support asynchronous responses, such as returning a task ID after a POST request so that the client can poll for the status of that task?
2. Can Flowise export the constructed flow as LangChain code? This would allow me to deploy the code as a separate service using server tools.

**Comment by HenryHengZJ** [comment]
If your Flowise instances are connecting to a single database (we support postgresql, mysql, sqlite), then yes changes made on 1 instance will be reflected to other as well.

1. We dont currently support async response yet, that is on the roadmap.
2. Same as well, on the roadmap

**Comment by BigFoxMedia** [comment]
Following! :-)
